### ISSUE TITLE
Material UI Examples

### ISSUE THREAD

**Issue by alexhawkins** [erikras/redux-form#542]
Does anyone have any examples using Redux Form with Material UI?

**Comment by luisrudge** [comment]
`<TextField {...this.props.fields.name} />` works perfectly.

**Comment by luisrudge** [comment]
@alexhawkins this is a real world exampe: `<TextField {...this.props.fields.name} />`.
It's from my own codebase. You just have to pass the field options to the input and everything will work. If it's not working, please create a jsdfiddle or codepen thing so I can help you out

**Comment by rhythnic** [comment]
I'm using it with Material UI as well.  I've had to wrap SelectField, RadioButton, and DatePicker to change the component Interface a bit.

```
class SelectFieldWrapper extends Component{
  onChange(evt, index, value) {
    if (this.props.onChange) {
      this.props.onChange(value);
    }
  }
  render() {
    return (
      <SelectField {...this.props} onChange={this.onChange.bind(this)}>
        {this.props.children}
      </SelectField>
    );
  }
}

class RadioButtonWrapper extends Component {
  onCheck(evt, value) {
    this.props.onChange(evt);
  }
  render() {
    return <RadioButton {...this.props} onCheck={this.onCheck.bind(this)}/>
  }
}

class DatePickerWrapper extends Component {
  onChange(evt, date) {
    if (this.props.onChange) {
      this.props.onChange(date);
    }
  }
  render() {
    return <DatePicker {...this.props} onChange={this.onChange.bind(this)}/>
  }
}
```

**Comment by luisrudge** [comment]
ohhhhh, yeah! I had to do that too! Sorry, I was only seeing the `TextField` input. 
But yeah, some form inputs will require you to mangle the `onChange` event a bit

**Comment by alexhawkins** [comment]
Awesome , thanks for the examples guys. I got Material UI working with Redux Forms last night. Also added custom validations to the fields. Here is the component I created. Hopefully this helps people that are interested in using Material UI with Redux Form. Note that I'm using FlexBoxGrid.css. Row Box Col are custom components I created, they are not part of Material UI.

```
import React, {Component, PropTypes} from 'react'
import {reduxForm} from 'redux-form'
import newAccountValidation from './newAccountValidation'

import Paper from 'material-ui/lib/paper'
import TextField from 'material-ui/lib/text-field'
import RaisedButton from 'material-ui/lib/raised-button'
import DropDownMenu from 'material-ui/lib/DropDownMenu'
import MenuItem from 'material-ui/lib/menus/menu-item'
import RadioButton from 'material-ui/lib/radio-button'
import RadioButtonGroup from 'material-ui/lib/radio-button-group'

import ActionLock from 'material-ui/lib/svg-icons/action/lock'
import CommunicationMailOutline from 'material-ui/lib/svg-icons/communication/mail-outline'
import SocialPerson from 'material-ui/lib/svg-icons/social/person'

import Row from 'shared/components/FlexboxGrid/Row'
import Col from 'shared/components/FlexboxGrid/Col'
import Box from 'shared/components/FlexboxGrid/Box'
import classes from './NewAccountForm.scss'

export const fields = [
  'first_name', 'last_name', 'email', 'password',
  'password2', 'company_name', 'phone_number',
  'website', 'address1', 'address2', 'city',
  'state', 'zip', 'feature_set', 'marketo_id'
]

class NewAccountForm extends Component {

  static propTypes = {
    account: PropTypes.object.isRequired,
    createAccount: PropTypes.func.isRequired,
    active: PropTypes.string,
    fields: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    resetForm: PropTypes.func.isRequired,
    submitting: PropTypes.bool.isRequired,
    dirty: PropTypes.bool.isRequired,
    resetForm: PropTypes.func.isRequired,
    invalid: PropTypes.bool.isRequired,
    pristine: PropTypes.bool.isRequired,
    valid: PropTypes.bool.isRequired
  };

  constructor(props) {
    super(props)
  }

  onSubmit(formData) {
    this.props.createAccount(formData)
  }

  render() {

    const {
      handleSubmit, resetForm, submitting, invalid, pristine,
      fields: {
        first_name, last_name, email, password,
        password2, company_name, phone_number,
        website, address1, address2,city, state,
        zip, feature_set, marketo_id
        },
      } = this.props

    return (
      <form autoComplete='off' className={classes['formContainer']} onSubmit={handleSubmit(this.onSubmit.bind(this))}>
        <Row>
          <Col xs={12} sm={12} md={6} lg={6} className={classes['formCardLeft']}>
            <Box>
              <Paper
                style={ {'padding': '0 1rem 3rem 2rem' } }
                zDepth={1}
                rounded>
                <Row>
                  <Col xs={1}>
                    <SocialPerson
                      color={'hsla(0, 0%, 46%, 1)'}
                      style={{'marginTop': '2.65rem'}}

                    />
                  </Col>
                  <Col xs={5} sm={5} md={5} lg={5}>
                    <Box>

                      <TextField {...first_name}
                        errorText={first_name.touched && first_name.error ? first_name.error : ''}
                        hintText='First Name'
                        floatingLabelText='First Name'
                        autoComplete='off'
                        style={ {'width':'90%'} }
                      />
                    </Box>
                  </Col>
                  <Col xs={6} sm={6} md={6} lg={6}>
                    <Box>
                      <TextField {...last_name}
                        errorText={last_name.touched && last_name.error  ? last_name.error : ''}
                        hintText='Last Name'
                        floatingLabelText='Last Name'
                        style={ {'width':'90%'} }
                      />
                    </Box>
                  </Col>
                </Row>
                <Row>

                  <Col xs={1}>
                    <CommunicationMailOutline
                      color={'hsla(0, 0%, 46%, 1)'}
                      style={{'marginTop': '2.65rem'}}
                    />
                  </Col>
                  <Col xs={11}>
                    <Box>
                      <TextField {...email}
                        errorText={email.touched && email.error  ? email.error : ''}
                        hintText='hans.gruber@shiphawk.com'
                        floatingLabelText='Email'
                        style={ {'width':'95%'} }
                      />
                    </Box>
                  </Col>
                </Row>
                <Row>
                  <Col xs={1}>
                    <Box>
                      <ActionLock
                        color={'hsla(0, 0%, 46%, 1)'}
                        style={{'marginTop': '2.65rem'}}
                      />
                    </Box>
                  </Col>
                  <Col xs={11}>
                    <Box>
                      <TextField {...password}
                        errorText={password.touched && password.error  ? password.error : ''}
                        hintText='Password'
                        floatingLabelText='Password'
                        style={ {'width': '95%'} }
                      />
                    </Box>
                  </Col>
                </Row>
                <Row>
                    <Col xs={1}>
                      <Box>
                        <ActionLock
                          color={'hsla(0, 0%, 46%, 0)'}
                          style={{'marginTop': '2.65rem'}}
                        />
                      </Box>
                    </Col>
                  <Col xs={11}>
                    <Box>
                      <TextField {...password2}
                        errorText={password2.touched && password2.error ? password2.error : ''}
                        hintText='Password Confirmation'
                        floatingLabelText='Password Confirmation'
                        style={ {'width':'95%', marginBottom: '2.5rem'} }
                      />
                    </Box>
                  </Col>
                </Row>
                <Row>
                  <Col xs={4}>
                    <Box>
                      <RadioButtonGroup {...feature_set} name='feature_set' defaultSelected='enterprise'>
                        <RadioButton
                          defaultChecked={true}
                          value='basic'
                          label='Pro'
                          style={{'paddingLeft':'0.75rem', 'paddingRight':'0.5rem', marginBottom: 15}}/>
                        <RadioButton
                          value='enterprise'
                          label='Enterprise'
                          style={{'paddingLeft':'0.75rem', marginBottom: 10}}/>
                      </RadioButtonGroup>
                    </Box>
                  </Col>
                  <Col xs={8}>
                    <Box>
                      <TextField {...marketo_id}
                        hintText='Marketo Lead ID'
                        floatingLabelText='Marketo Lead ID'
                        style={ {'width':'97.5%'} }
                      />
                    </Box>
                  </Col>
                </Row>
              </Paper>
            </Box>
          </Col>

          <Col xs={12} sm={12} md={6} lg={6} className={classes['formCardRight']}>
            <Box>
              <Paper
                style={ { 'padding': '0 1.5rem  3rem 2rem' } }
                zDepth={ 1 }
                rounded>

                <Col xs={12}>
                  <Box>
                    <TextField {...company_name}
                      errorText={company_name.touched && company_name.error ? company_name.error : ''}
                      hintText='Company Name'
                      floatingLabelText='Company Name'
                      style={ {'width':'97.5%'} }
                    />
                  </Box>
                </Col>
                <Col xs={12}>
                  <Box>
                    <TextField {...phone_number}
                      errorText={invalid && phone_number.touched && phone_number.error  ? phone_number.error : ''}
                      hintText='3105501940'
                      floatingLabelText='Phone Number'
                      style={ {'width':'97.5%'} }
                    />
                  </Box>
                </Col>

                <Col xs={12}>
                  <Box>
                    <TextField {...website}
                      hintText='https://wwww.shiphawk.com'
                      floatingLabelText='Website'
                      style={ {'width':'97.5%'} }
                    />
                  </Box>
                </Col>
                <Col xs={12}>
                  <Box>
                    <TextField {...address1}
                      hintText='Address 1'
                      floatingLabelText='Address 1'
                      style={ {'width':'97.5%'} }
                    />
                  </Box>
                </Col>

                <Col xs={12}>
                  <Box>
                    <TextField {...address2}
                      hintText='Address 2'
                      floatingLabelText='Address 2'
                      style={ {'width': '97.5%'} }
                    />
                  </Box>
                </Col>
                <Row>
                  <Col xs={12} sm={4} md={4} lg={4}>
                    <Box>
                      <TextField {...city}
                        hintText='City'
                        floatingLabelText='City'
                        primary
                        style={ {'width':'97.5%'} }
                      />
                    </Box>
                  </Col>

                  <Col xs={12} sm={4} md={4} lg={4}>
                    <Box>
                      <TextField {...state}
                        hintText='State'
                        floatingLabelText='State'
                        style={ {'width': '97.5%'} }
                      />
                    </Box>
                  </Col>
                  <Col xs={12} sm={4} md={4} lg={4}>
                    <Box>
                      <TextField {...zip}
                        hintText='Zip'
                        floatingLabelText='Zip'
                        style={ {'width':'92.5%'} }
                      />
                    </Box>
                  </Col>
                </Row>
              </Paper>
            </Box>
          </Col>
        </Row>

        <div className='row end-xs'>
          <Col xs={2} sm={4} md={7} lg={8}>

          </Col>
          <Col xs={3} sm={3} md={2} lg={1}>
            <Box>
              <RaisedButton
                label='Reset'
                primary={true}
                style={ {'width': '100%'} }
                onTouchTap={resetForm}
              />
            </Box>
          </Col>
          <Col xs={6} sm={4} md={3} lg={2}>
            <Box>
              <RaisedButton
                label='Create Account'
                secondary={true}
                style={ {'width': '100%'} }
                type='submit'
              />
            </Box>
          </Col>
        </div>
      </form>
    )
  }
}

export default reduxForm({
  form: 'new-account',
  fields,
  validate: newAccountValidation
})(NewAccountForm)

```

**Comment by erikras** [comment]
Thank you so much for that, @alexhawkins! That's going to help a lot of people.

**Comment by hourliert** [comment]
Thanks @alexhawkins !!

**Comment by alexhawkins** [comment]
No problem @erikras happy to help. @rhythnic For Radio Button, I haven't found the need to create a wrapper. This works fine for me:

```
 <RadioButtonGroup name='feature_set' defaultSelected='basic'>
                          <RadioButton {...feature_set}
                            value='basic'
                            label='Basic'
                            style={{'paddingLeft': '0.25rem', 'marginTop': '1rem', 'marginBottom':'1rem'}}/>
                          <RadioButton  {...feature_set}
                            value='plus'
                            label='Plus'
                            style={{'paddingLeft': '0.25rem', 'marginTop': '1rem', 'marginBottom':'1rem'}}/>
                          <RadioButton {...feature_set}
                            value='pro'
                            label='Pro'
                            style={{'paddingLeft': '0.25rem', 'marginTop': '1rem', 'marginBottom':'1rem'}}/>
                          <RadioButton {...feature_set}
                            value='enterprise'
                            label='Enterprise'
                            style={{'paddingLeft': '0.25rem', 'marginTop': '1rem', 'marginBottom':'1rem'}}/>
 </RadioButtonGroup>
```

**Comment by marfillaster** [comment]
@alexhawkins 

here's my working version
```
<RadioButtonGroup name="toggle" defaultSelected="yes"
          onChange={(e, v) => toggle.onChange(v)}
          >
          <RadioButton {...toggle}
            label="No"
            value="no"
            onChange={noop => noop}
          />
          <RadioButton {...toggle}
            label="Yes"
            value="yes"
            onChange={noop => noop}
          />
        </RadioButtonGroup>
```

**Comment by newsiberian** [comment]
Hello, could anyone provide example with [`<Toggle>`](http://www.material-ui.com/#/components/toggle)?
Doing it by simple way:
```js
<Toggle
  {...allowGuestCheckout}
  label={t("shopEditForm.allowGuestCheckout")}
/>
```
I've got: ```Warning: Failed propType: Invalid prop `switched` of type `object` supplied to `EnhancedSwitch`, expected `boolean`. Check the render method of `Toggle`.```

**Comment by gkkirsch** [comment]
Anyone else experiencing problems with BLUR and FOCUS actions being fired at the same time when you click on a Material Ui TextField? This problem only occurs when I am trying to overwrite the default theme using childContext.  If I follow what Material Ui says to do here [http://www.material-ui.com/#/customization/themes](url) my redux form fields hate themselves.

**Comment by antoinerousseau** [comment]
thanks @rhythnic !

**Comment by clarkie** [comment]
@rhythnic Thanks for the pointers. 

Is this still the standard way of handling these or have recent releases 'fixed' this? I'm well behind on material-ui@0.14.4 and redux-form@4.2.2

**Comment by rhythnic** [comment]
Material-UI RadioButton API changed from onCheck to onChange, so @alexhawkins is right in that they don't need to be wrapped.  I'm still wrapping Checkbox to convert onCheck to onChange.  I also wrap DatePicker and SelectField, as shown above, but I'm not sure if Select needs to be wrapped.  I haven't tried unwrapping it yet.  I don't think the updates change much except for the Material UI import paths.

**Comment by erikras** [comment]
Just thought I would mention in this thread that `v6` already has [a working Material UI example](http://redux-form.com/6.0.0-alpha.13/examples/material-ui/).

**Comment by itaylorweb** [comment]
@erikras Can you provide an example with AutoComplete?  The result I get is the search term rather than the selected item.  For example if I have a dataSource of countries and I type 'uk' into the input and select 'United Kingdom' from the list, the value of the field remains 'uk' not 'United Kingdom'

**Comment by pieval** [comment]
@erikras Does your example prodvided above in the v6.XX documentation already work (I'm using react 15.2.1, redux 3.5.2, redux-form 6.0.0rc3) using standard material-ui component ? I mean using this syntax 
```javascript
<Field name="firstName" component={firstName => 
          <TextField hintText = "First Name" 
            floatingLabelText="First Name"
            errorText = {firstName.touched && firstName.error}
            {...firstName} 
          />
```
Or does I have to use your redux-form-material-ui project ?

Right now, when I use the syntax above, my TextFiled seems not be connected to redux state.

**Comment by pieval** [comment]
Ok, found ! I have to spread ```firstName.input```.

**Comment by GuillaumeCisco** [comment]
I've just managed to make the datePicker component clearable with the new Field API.
If you're interested, you can see code here:
https://github.com/callemall/material-ui/issues/1529#issuecomment-235516337

**Comment by betoharres** [comment]
could someone paste a example using material-ui in the entire form? 
my problem is that the docs does not show a **complete** material-ui example, they jumped the last part when you fire handleSubmit using a `form` tag, which for me it's too much magic. 

I'm used to collect the redux state and fire a `redux-thunk` to the api, but the example shown makes impossible to understand( for me ) how I could achieve the same thing using this library.

I'd happily make a PR so anyone else run in the same problem. Many thanks in advance, loved the library ❤️

**Comment by gustavohenke** [comment]
Hi @betoharres, what's your problem with the Material UI example?
I see you're struggling with it, but I don't understand what parts should be made clearer.

**Comment by betoharres** [comment]
@gustavohenke Hello, *tchê*! hehe
I'm trying to find another way around of submitting a form. 

Instead of using`<form onSubmit={handleSubmit}>`
I'd like to do something like this:
`<RaisedButton label='submit' onClick={handleSubmit} />`
and then in my container component I'd do the same thing as shown in the `Getting Started` section:
```es6
class ContactPage extends React.Component {
  handleSubmit = (values) => {
    // Do something with the form values
    console.log(values);
  }
  render() {
    return (
      <ContactForm onSubmit={this.handleSubmit} />
    );
  }
}
```

I know I could reach the state, but instead I was looking for send the form values to the `handleSubmit` as arguments without needing to search throw them in the state.

Does this help? *Valeu* 😄

**Comment by gustavohenke** [comment]
It seems like what you want is the [remote submit example](http://redux-form.com/6.5.0/examples/remoteSubmit/)

**Comment by betoharres** [comment]
@gustavohenke Thanks a bunch! I'll look into that 🎉

**Comment by betoharres** [comment]
I'm sorry to bother again but I had no success with Remote Submit example =[

I follow this workflow of setting up a container component where all the logic lives, and then it renders a stateless component which is just a function returning the UI it's actions.

The problem is that the example shows the reverse of how I set up my code, connecting the stateless component to the redux store( making it a stateful component ) so it can use the `dispatch` to fire `redux-form` own action called `submit` and there's no way to send the values to the container component since I need to "connect" the stateless component with `redux-form` like: `reduxForm()(StatelessForm)` ... (**checkmate**).
Any help with this? I'm only asking help there because I burned all my brain on this 😕  Thanks in advance
